### DISCUSSION TITLE
Do you have vite hot reload with css modules?

### DISCUSSION THREAD

**Discussion by tombohub** [remix-run/remix#8781] [Q&A] [answered]
When I change css module it doesn't reload.

Is that normal or a bug?

**Comment by kiliman** [comment, marked as answer]
I found that Vite HMR doesn't work for CSS if you have hydration errors. Remix and React 18.2 have issues with hydration if you have browser extensions installed that modify the DOM. 

I find that installing React 18.3 Canary fixes the hydration problem. Even Next.js uses canary so I don't think that's a concern. 

See my Remix+Vite template for guidance. 

https://github.com/kiliman/remix-vite-template

**Reply by tombohub** [reply]
For some reason now it's working and I don't remember touching anything.
